core/src/main/java/com/onelogin/saml2/logout/LogoutRequest.java

```diff
@@ -334,16 +334,16 @@ public Boolean isValid() throws Exception {
 				if (signAlg == null || signAlg.isEmpty()) {
 					signAlg = Constants.RSA_SHA1;
 				}
-				String relayState = request.getParameter("RelayState");
+				String relayState = request.getEncodedParameter("RelayState");
 
-				String signedQuery = "SAMLRequest=" + Util.urlEncoder(request.getParameter("SAMLRequest"));
+				String signedQuery = "SAMLRequest=" + request.getEncodedParameter("SAMLRequest");
 
 				if (relayState != null && !relayState.isEmpty()) {
-					signedQuery += "&RelayState=" + Util.urlEncoder(relayState);
+					signedQuery += "&RelayState=" + relayState;
 				}
 
-				signedQuery += "&SigAlg=" + Util.urlEncoder(signAlg);
-
+				signedQuery += "&SigAlg=" + request.getEncodedParameter("SigAlg", signAlg);
```

Comment:
This still allows an not specifying the `SigAlg` parameter. The spec says this MUST be included. I'd prefer checking and requiring that here, rather than falling back to the default. @pitbulk, any potential reason not to?